### DISCUSSION TITLE
Shutting down the server from another thread

### DISCUSSION THREAD

**Discussion by tommitytom** [uNetworking/uWebSockets#1139] [Q&A] [answered]
I'm running a server in its own thread.  It has no interaction with any other threads, however I would like to know if there is a way to shut down/destroy the server from my main thread

**Comment by ghost** [comment, marked as answer]
You cannot shut down the server from another thread but you can Loop::defer the shutting down from any thread:

```
// From the thread where the loop runs we get and store the loop ptr
Loop *loopToShutdown = Loop::get();

// Loop * is directly castable to us_loop_t *

// From any thread
loopToShutdown->defer([]() {
  us_listen_socket_close(...)
});
```
